language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/FunctionCallExpressionTypeFinder.java

```diff
@@ -345,6 +349,35 @@ public void visit(ImplicitNewExpressionNode implicitNewExpressionNode) {
                 .stream().findFirst().ifPresent(this::checkAndSetTypeResult);
     }
 
+    @Override
+    public void visit(FunctionDefinitionNode node) {
+        semanticModel.symbol(node)
+                .flatMap(SymbolUtil::getTypeDescriptor)
+                .ifPresent(this::checkAndSetTypeResult);
+    }
+
+    @Override
+    public void visit(FunctionBodyBlockNode node) {
+        node.parent().accept(this);
+    }
+
+    @Override
+    public void visit(ReturnStatementNode returnStatementNode) {
+        this.semanticModel.typeOf(returnStatementNode).ifPresent(this::checkAndSetTypeResult);
+        if (resultFound) {
+            return;
+        }
+
+        // Get function type symbol and get return type descriptor from it
+        returnStatementNode.parent().accept(this);
+        if (resultFound && returnTypeSymbol.typeKind() == TypeDescKind.FUNCTION) {
+            FunctionTypeSymbol functionTypeSymbol = (FunctionTypeSymbol) returnTypeSymbol;
+            functionTypeSymbol.returnTypeDescriptor().ifPresentOrElse(this::checkAndSetTypeResult, this::resetResult);
```

Comment:
Isn't it better if we move this logic to the function definition node itself.  As the return type  of function represents  the particular  function, I feel that this logic does not belong here.

Reply:
I thought of having this logic in the `FunctionDefinitionNode` visit method. Then decided not to. 
```ballerina
function myFunction(int x = fn1()) returns int {
   return undefinedFunction();
}
```
Can we guarantee that the `visit(FunctionDefinitionNode)` is being called by via `function call expr -> return statement -> function body -> function definition` hierarchy only?
What if we wanted to find the return type of `fn1()`? (For the case of `fn1()`, I think it is covered by a function parameter node visit method. Not sure. But it's the idea.) Therefore, to keep the logic clean, I thought of going with the current approach. WDYT?